hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -554,6 +564,11 @@ private CompletableFuture<Message> writeStateMachineData(
     CompletableFuture<ContainerCommandResponseProto> writeChunkFuture =
         CompletableFuture.supplyAsync(() -> {
           try {
+            try {
+              checkContainerHealthy(write.getBlockID().getContainerID());
```

Comment:
IMO, If statemachine itself is unhealthy, we should reject any further writeStateMachineData(), so no need check for specific container. This is valid for applyTransaction() but not here.

Reply:
This function gets called on follower as well. We should not block other container writes. This will lead to a lot of container getting impacted because of one container failure.

Reply:
Since we are returning an error response, apply transaction will still go through there we wouldn't find the to be unhealthy container we might end applying a putBlock transaction without actually writing data.

Reply:
as discussed, we should avoid additional request for the pipeline in writeData case. Application Transaction getting continue for the data already present is fine.